scripts/create_joined_view.py

```diff
@@ -0,0 +1,391 @@
+import argparse
+import logging
+import logging.handlers
+import sys
+from arcgis.gis import GIS
+from arcgis.apps.workforce.project import Project
+from arcgis.features import FeatureLayerCollection
+
+
+def create_joined_view(gis, source_layer, join_layer, primary_key_field, foreign_key_field, name, source_fields,
+                       join_fields):
+    """
+    Create a joined layer view between 2 layers
+    :param gis: The gis to create the layer with
+    :param source_layer: The source layer to join
+    :param join_layer: The layer to join with the source layer
+    :param primary_key_field: The primary key field in the source layer
+    :param foreign_key_field: The foreign key field in the join layer
+    :param name: The name of the new layer that will be created
+    :param source_fields: The list of field configuration objects in the source layer to keep in the resulting joined layer
+    :param join_fields: The list of field configuration objects in the join layer to keep in the resulting joined layer
+    :return: The new item
+    """
+    new_item = gis.content.create_service(
+        name=name,
+        is_view=True,
+        create_params={
+            "currentVersion": 10.2,
+            "serviceDescription": "",
+            "hasVersionedData": False,
+            "supportsDisconnectedEditing": False,
+            "hasStaticData": True,
+            "maxRecordCount": 2000,
+            "supportedQueryFormats": "JSON",
+            "capabilities": "Query",
+            "description": "",
+            "copyrightText": "",
+            "allowGeometryUpdates": False,
+            "syncEnabled": False,
+            "editorTrackingInfo": {
+                "enableEditorTracking": False,
+                "enableOwnershipAccessControl": False,
+                "allowOthersToUpdate": True,
+                "allowOthersToDelete": True
+            },
+            "xssPreventionInfo": {
+                "xssPreventionEnabled": True,
+                "xssPreventionRule": "InputOnly",
+                "xssInputRule": "rejectInvalid"
+            },
+            "tables": [
+
+            ],
+            "name": f"{name}"
+        }
+    )
+    fc = FeatureLayerCollection.fromitem(new_item)
+    layer_def = generate_layer_definition(source_layer, join_layer, primary_key_field, foreign_key_field, name,
+                                          source_fields, join_fields)
+    fc.manager.add_to_definition(layer_def)
+    return new_item
+
+
+def assignment_type_fields(use_joined_name=False):
+    """
+    Creates the list of field configuration objects to use in the joined layer
+    :param use_joined_name: Determines if the source field should be the same as the name (used when creating a view on a view)
+    :return: List of field configuration objects
+    """
+    fields = [
```

Comment:
What would you think about defining each of these field config objects outside of the original functions, have them be representative of the state of the field lists that should be used, and just have a function to update the requisite? It would be slightly more efficient and reduce code repetition

So I'm thinking it becomes like: 
```
assignments_to_types = create_joined_view(gis,
                                              project.assignments_layer,
                                              project.assignment_types_table,
                                              project._assignment_schema.assignment_type,
                                              project._assignment_types.global_id,
                                              f"{project.title}_IntermediateView0",
                                              assignment_fields,
                                              assignment_type_fields)
use_joined_names(assignment_type_fields)
```
etc 

This makes more sense to me because it's not like you need to go back and forth when performing each join (i.e. first iteration, assignment types don't need joined name, then need joined name, then don't need it) - it's unidirectional

Reply:
I made this change